public/app/features/plugins/datasource_srv.ts

```diff
@@ -324,6 +325,11 @@ export class DatasourceSrv implements DataSourceService {
       };
     });
   }
+
+  async reload() {
+    const settings = await getBackendSrv().get('/api/frontend/settings');
+    this.init(settings.datasources, settings.defaultDatasource);
+  }
```

Comment:
There is similar code in a redux action:
https://github.com/grafana/grafana/blob/main/public/app/features/datasources/state/actions.ts#L286

That one mutates config -- perhaps we should do the same here and then have the reduc action call `reload()`?

Reply:
@ryantxu good call: https://github.com/grafana/grafana/pull/44217/commits/38b3adcfa3c6aeb4dcecdc281def2b27202f4a19